File: Itido-BMS/RentAPI/Rent/Models/TimePlanning/WorkRegistration.cs
using System;
using System.Linq.Expressions;
namespace Rent.Models.TimePlanning
{
    public class WorkRegistration
    {
        public int ID { get; set; }
        public int WorkID { get; set; }
        public DateTime Date { get; set; }
        public int UserID { get; set; }
        public short? StartTimeMins { get; set; }
        public short? EndTimeMins { get; set; }
        public short? BreakMins { get; set; }
        public int? RequestID { get; set; }

        public virtual Work Work { get; set; }

        public virtual Request Request { get; set; }
        public virtual User User { get; set; }

        public static Expression<Func<WorkRegistration, dynamic>> BasicDTO()
        {
            return v => v != null ?
            new
            {
                v.WorkID,
                v.Date,
                v.StartTimeMins,
                v.EndTimeMins,
                //Request = (v.Request != null && v.Request.Creator != null && v.Request.Subject != null) ? Request.BasicDTO(requester, requesterRole).Compile()(v.Request, v.Request.Creator) : null,
            } : null;
        }

    }


}

File: Itido-BMS/dart/bms_dart/lib/src/models/work_registration.dart
import 'package:bms_dart/models.dart';
import 'package:bms_dart/src/models/date_time_converter.dart';
import 'package:flutter/material.dart';

class WorkRegistration {
  // int workId;
  DateTime date;
  int startTimeMins;
  int endTimeMins;
  int breakMins;
  Request request;

  WorkRegistration({
    // this.workId,
    this.date,
    this.startTimeMins,
    this.endTimeMins,
    this.breakMins,
  });

  factory WorkRegistration.fromJson(json, {rootJson, path}) {
    if (json == null) return null;
    return WorkRegistration._fromJson(json, rootJson ?? {}, path ?? '');
  }

  WorkRegistration._fromJson(json, rootJson, path)
      : this.date = toDateTime(json['date']),
        this.startTimeMins = json['startTimeMins'],
        this.endTimeMins = json['endTimeMins'],
        this.breakMins = json['breakMins'],
        this.request = Request.fromJson(
          json['request'] ?? rootJson['${path}_request'],
          rootJson: rootJson,
          path: '${path}_request',
        );

  Map<String, dynamic> toMap() => {
        'date': this.date.toString(),
        'startTimeMins': this.startTimeMins,
        'endTimeMins': this.endTimeMins,
        'breakMins': this.breakMins,
      };
}
